code/modules/games/cards.dm

```diff
@@ -0,0 +1,443 @@
+			qdel(src)
+			H.update_icon()
+			return
+		else
+			to_chat(user,"<span class = 'notice'>You cannot mix cards from other deck!</span>")
+			return
+	..()
+
+/obj/item/weapon/deck/attack_self()
+	deckshuffle()
+
+/obj/item/weapon/deck/verb/verb_shuffle()
+	set category = "Object"
+	set name = "Shuffle"
+	set desc = "Shuffle the cards in the deck."
+	set src in view(1)
+	deckshuffle()
+
+/obj/item/weapon/deck/proc/deckshuffle()
+	var/mob/living/user = usr
+	if (cooldown < world.time - 50)
+		cards = shuffle(cards)
+		user.visible_message("<span class = 'notice'>[user] shuffles [src].</span>")
+		playsound(user, 'sound/items/cardshuffle.ogg', 50, 1)
+		cooldown = world.time
+
+
+/obj/item/weapon/deck/MouseDrop(atom/over_object) // Code from Paper bin, so you can still pick up the deck
+	var/mob/M = usr
+	if(M.restrained() || M.stat || !Adjacent(M))
+		return
+
+	if(over_object == M)
+		M.put_in_hands(src)
+
+	else if(istype(over_object, /obj/screen))
+		switch(over_object.name)
+			if("r_hand")
+				if(!remove_item_from_storage(M))
+					M.unEquip(src)
+				M.put_in_r_hand(src)
+			if("l_hand")
+				if(!remove_item_from_storage(M))
+					M.unEquip(src)
+				M.put_in_l_hand(src)
+
+	add_fingerprint(M)
+
+/obj/item/weapon/pack
+	name = "Card Pack"
+	desc = "For those with disposable income."
+
+	icon_state = "card_pack"
+	icon = 'icons/obj/playing_cards.dmi'
+	w_class = WEIGHT_CLASS_TINY
+	var/list/cards = list()
+	var/parentdeck = null // For future card pack that need to be compatible with eachother i.e. cardemon
+
+
+/obj/item/weapon/pack/attack_self(var/mob/user as mob)
+	user.visible_message("<span class = 'notice'>[name] rips open the [src]!</span>")
+	var/obj/item/weapon/hand/H = new()
+
+	H.cards += cards
+	cards.Cut();
+	user.drop_item()
+	qdel(src)
+
+	H.update_icon()
+	user.put_in_active_hand(H)
+
+/obj/item/weapon/hand
+	name = "hand of cards"
+	desc = "Some playing cards."
+	icon = 'icons/obj/playing_cards.dmi'
+	icon_state = "empty"
+	w_class = WEIGHT_CLASS_TINY
+
+	var/concealed = FALSE
+	var/list/cards = list()
+	var/parentdeck = null
+
+/obj/item/weapon/hand/verb/discard()
+
+	set category = "Object"
+	set name = "Discard"
+	set desc = "Place cards from your hand in front of you."
+
+	var/i
+	var/maxcards = min(cards.len,5) // Maximum of 5 cards at once
+	var/discards = input("How many cards do you want to discard? You may discard up to [maxcards] card(s)") as num
+	if(discards > maxcards)
+		return
+	for	(i = 0;i < discards;i++)
+		var/list/to_discard = list()
+		for(var/datum/playingcard/P in cards)
+			to_discard[P.name] = P
+		var/discarding = input("Which card do you wish to put down?") as null|anything in to_discard
+
+		if(!discarding || !to_discard[discarding] || !usr || !src) return
+
+		var/datum/playingcard/card = to_discard[discarding]
+		to_discard.Cut()
+
+		var/obj/item/weapon/hand/H = new(src.loc)
+		H.cards += card
+		cards -= card
+		H.parentdeck = src.parentdeck
+		H.concealed = FALSE
```

Comment:
Not sure, but will this reveal the new hand?

Reply:
Intended as it "play" the card